docs/reference/mapping/runtime.asciidoc

```diff
@@ -747,6 +747,243 @@ address.
+definition into the mappings of the new index. It's that simple!
+
+[source,console]
+----
+PUT my-index-00001/
+{
+  "mappings": {
+    "properties": {
+      "timestamp": {
+        "type": "date"
+      },
+      "temperature": {
+        "type": "long"
+      },
+      "voltage": {
+        "type": "double"
+      },
+      "node": {
+        "type": "keyword"
+      },
+      "voltage_reading": {
+        "type": "long",
+        "script": {
+          "source": """
+            for (double v : doc['voltage'])
+            emit((long)(v * params['multiplier']))
+            """,
+          "params": {
+            "multiplier": 4
+          }
+        }
+      }
+    }
+  }
+}
+----
+
+Bulk index some sample data from your sensors into the `my-index-00001` index:
+
+[source,console]
+----
+POST my-index-00001/_bulk?refresh=true
+{ "index": {}}
+{ "timestamp": 1516729294000, "temperature": 200, "voltage": 5.2, "node": "a"}
+{ "index": {}}
+{ "timestamp": 1516642894000, "temperature": 201, "voltage": 5.8, "node": "b"}
+{ "index": {}}
+{ "timestamp": 1516556494000, "temperature": 202, "voltage": 5.1, "node": "a"}
+{ "index": {}}
+{ "timestamp": 1516470094000, "temperature": 198, "voltage": 5.6, "node": "b"}
+{ "index": {}}
+{ "timestamp": 1516383694000, "temperature": 200, "voltage": 4.2, "node": "c"}
+{ "index": {}}
+{ "timestamp": 1516297294000, "temperature": 202, "voltage": 4.0, "node": "c"}
+----
+// TEST[continued]
+
+You can now retrieve calculated values in a search query, and find documents
+based on precise values. The following range query returns all documents where
+the calculated `voltage_reading` is greater than or equal to `10`, but less
+than or equal to `16`:
+
+[source,console]
+----
+GET my-index-00001/_search
+{
+  "query": {
+    "range": {
+      "voltage_reading": {
+        "gte": 16,
+        "lte": 20,
+        "boost": 1.0
+      }
+    }
+  }
+  , "size": 2
+}
+----
+// TEST[continued]
+
+Even though `voltage_reading` isn't included in the response, the documents
+that match the range query are returned based on the calculated value of the
+included script:
+
+[source,console-result]
+----
+{
+  ...
+  "hits" : {
+    "total" : {
+      "value" : 4,
+      "relation" : "eq"
+    },
+    "max_score" : 1.0,
+    "hits" : [
+      {
+        "_index" : "my-index-00001",
+        "_id" : "fYR4qHgBdg9xpPrUEz-q",
+        "_score" : 1.0,
+        "_source" : {
+          "timestamp" : 1516729294000,
+          "temperature" : 200,
+          "voltage" : 5.2,
+          "node" : "a"
+        }
+      },
+      {
+        "_index" : "my-index-00001",
+        "_id" : "f4R4qHgBdg9xpPrUEz-q",
+        "_score" : 1.0,
+        "_source" : {
+          "timestamp" : 1516556494000,
+          "temperature" : 202,
+          "voltage" : 5.1,
+          "node" : "a"
+        }
+      }
+    ]
+  }
+}
+----
+// TESTRESPONSE[s/\.\.\./"took" : $body.took,"timed_out" : $body.timed_out,"_shards" : $body._shards,/]
```

Comment:
You can skip this if you do `filter_path=hits`. I'm not sure if that's a good idea or not, but its a choice.

Reply:
Yeah, you're right! I'll add this to the request and remove this more complex replacement pattern.